Core/Core.cpp

```diff
@@ -188,6 +197,11 @@ void Core_RunLoop() {
 			Sleep(sleepTime);
 		if (!windowHidden) {
 			GPU_SwapBuffers();
+
+			// Keep the system awake for longer than normal for cutscenes and the like.
+			if (time_now_d() < lastActivity + ACTIVITY_IDLE_TIMEOUT) {
+				SetThreadExecutionState(ES_SYSTEM_REQUIRED | ES_DISPLAY_REQUIRED);
```

Comment:
I wonder if this function is expensive? Seems like it should be easy to avoid calling it on every frame. Although, calling even an expensive function once per frame shouldn't be a big deal..